packages/ocl/src/Util.js

```diff
@@ -0,0 +1,106 @@
+import { composeNextState } from '@openfn/language-common';
+
+export function buildUrl(configuration, options) {
```

Comment:
i'm skeptical of the value of this. seems like a lot of code and i'd prefer to see string interpolation.

Reply:
string interpolation will have one huge string
```
const url = `${baseUrl}/${ownerType}/${ownerId}/${repository}/${repositoryId}/${version}/${expansions}/${expansionId}/${content}`;
```
 and all these values 👇🏽   are optional, so if a user does not pass them you have to write a logic that will remove `undefined` values from the string 
 ```
ownerId,
ownerType,
repository,
repositoryId,
version,
expansions,
expansionId,
```

The current implementation is readable and cleans up `undefined` optional values.
So i would vote for `const url = urlParts.join('/')` and not `const url = ${....}`